app/src/main/java/org/ole/planet/myplanet/ui/viewer/ResourceViewerViewModel.kt

```
@@ -40,4 +50,23 @@ class ResourceViewerViewModel @Inject constructor(
+        if (!FileUtils.checkFileExist(context, url)) {
+            DownloadUtils.openDownloadService(context, arrayListOf(url), false)
+        }
+    }
+
+    suspend fun extractPdfText(file: File): String = withContext(dispatcherProvider.io) {
+        try {
+            PDFBoxResourceLoader.init(context)
+            PDDocument.load(file).use {
+                PDFTextStripper().getText(it).trim()
+            }
+        } catch (e: Exception) { "" }
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)